src/dbnode/client/session.go

```diff
@@ -1988,18 +1957,8 @@ func (s *session) streamBlocksMetadataFromPeers(
 			}
 			for condition() {
 				var err error
-				switch version {
-				case FetchBlocksMetadataEndpointV1:
-					currPageToken, err = s.streamBlocksMetadataFromPeer(namespace, shardID,
-						peer, start, end, currPageToken, metadataCh, progress)
-				case FetchBlocksMetadataEndpointV2:
-					currPageToken, err = s.streamBlocksMetadataFromPeerV2(namespace, shardID,
-						peer, start, end, currPageToken, metadataCh, resultOpts, progress)
-				default:
-					// Should never happen - we validate the version before this function is
-					// ever called
-					err = xerrors.NewNonRetryableError(errInvalidFetchBlocksMetadataVersion)
-				}
+				currPageToken, err = s.streamBlocksMetadataFromPeerV2(namespace, shardID,
```

Comment:
nit: Maybe rename `streamBlocksMetadataFromPeerV2` to just `streamBlocksMetadataFromPeer`?